File: C/LinearSearch.c
#include<stdio.h>

int linearSearch(int arr[],int n,int x){
int j;

for(j=0;j<n;j++){
    if(arr[j]==x){
        return j;
    }
}
 j=-1;
    return j;

}

int main(){
    int n,x,i;
    int arr[100];
    printf("Enter size of array:\n");
     scanf("%d",&n);
     printf("Enter Array element:\n");

     for(i=0;i<n;i++){
        scanf("%d",&arr[i]);
     }
     printf("Enter your Search Item:\n");
     scanf("%d",&x);

     printf("Your Search Result is:\n");
     int result=linearSearch(arr,n,x);
     if(result!=-1){
        printf("%d",arr[result]);
     }else{
        printf("Not found");
     }

     /*
     This algorithm Time complexity :O(n) and Space complexity O(n);


     */
}

File: kotlin/src/linearSearch.kt
fun main(args:Array<String>){
    print("Enter your Array size:")
    var n= readLine()!!.toInt()

    var array=Array(n){0}
    for(i in 0 until n)
        array[i]= readLine()!!.toInt()

    print("Which number you want search? :")
    var key= readLine()!!.toInt()
    if(linearSearch(array,key)==1)
        print("Found $key")
    else
        print("Not found")
}
/*
2,4,5,7,8
key=5
 */



fun linearSearch(arr:Array<Int>,key:Int):Int{
    var n=arr.size
    var found=-1
    for(i in 0 until n)
        found = if (arr[i]==key)
            1
        else
            -1



   return found
}